### ISSUE TITLE
Shuffling data by default in learner.train

### ISSUE THREAD

**Issue by desilinguist** [EducationalTestingService/skll#189]
Currently, we shuffle the training data by default in `learner.train`. This leads to the issue that the results for the exact same data are different between SKLL and Scikit-Learn since the latter doesn't do any shuffling. Here's a [gist](https://gist.github.com/desilinguist/c70d6b62a8168fc796c2) illustrating this issue. 

I think we should:
1. Make `shuffle=False` by default. If the user does a grid search but doesn't set `shuffle` to True, then we should do the shuffling anyway and print out a note saying that's what we are doing.
2. Expose the `shuffle` parameter in the config file just like `grid_search`.
